src/ucs/sys/sock.h

```diff
@@ -539,6 +539,25 @@ int ucs_sockaddr_cmp(const struct sockaddr *sa1,
 int ucs_sockaddr_ip_cmp(const struct sockaddr *sa1, const struct sockaddr *sa2);
 
 
+/**
+ * Checks if two IP addresses are in the same subnet
+ *
+ * @param [in]  sa1         Pointer to sockaddr structure #1.
+ * @param [in]  sa2         Pointer to sockaddr structure #2.
+ * @param [in]  prefix_len  Subnet prefix length in bits.
+ * @param [out] is_same_p   Pointer to a boolean value indicating if both
+ *                          addresses are in the same subnet.
+ *
+ * @return UCS_OK on success or one of the following errors:
+ *         UCS_ERR_INVALID_PARAM - wrong or mismatched address families.
+ *         UCS_ERR_EXCEEDS_LIMIT - prefix_len is too big (exceeds size of
+ *                                 address).
+ */
+ucs_status_t ucs_sockaddr_is_same_subnet(const struct sockaddr *sa1,
```

Comment:
maybe just return if same subnet or not?
if there is any kind of error in the parameters, can return 0
if prefix is longer than address length, can check according to the actual address length